rrule/src/parser/content_line/content_line_parts.rs

```diff
@@ -0,0 +1,117 @@
+use crate::parser::{regex::get_property_name, ParseError};
+
+use super::PropertyName;
+
+#[derive(Debug, PartialEq, Clone)]
+pub(crate) struct ContentLineCaptures {
+    pub property_name: PropertyName,
+    pub parameters: Option<String>,
+    pub value: String,
+}
+
+/// Get the property name, parameters and values of a content line.
+pub(crate) fn get_content_line_parts(val: &str) -> Result<ContentLineCaptures, ParseError> {
```

Comment:
Cannot it be a `impl FromStr`?

Reply:
Yes, you are right, it _can_ be. I am a little bit inconsistent in my coding when choosing between `impl FromStr for Type` and `fn method(&str) -> Type`. Sometimes I want to add some docs to the implementation and give the method a specific name and therefore decide to go for `fn method..`, other times it feels more natural to implement `FromStr`. 

Do you follow any specific rules when choosing between the two approaches?

Reply:
If there is something in the standard library, I use that, because it's well known to developers and the behavior is exactly as they expected. That's it for me.

Reply:
Having changed `ContentLineCaptures` to contain `&str` references it is not possible to implement `FromStr`, because the `FromStr` trait does not allow tying the input lifetime to the output lifetime. Anonymous lifetime doesn't seem to work either. Ref: https://stackoverflow.com/questions/28931515/how-do-i-implement-fromstr-with-a-concrete-lifetime

Reply:
Yes, you are right. Then I suggest an `impl ContentLineCaptures` with a method like `from_str` or `from_blahblah_str`. Same for `ParsedDateString`.

Reply:
Completely agree 👍